### DISCUSSION TITLE
converting LaTeX to docx with equation and figure numbering and references

### DISCUSSION THREAD

**Discussion by tpapp** [jgm/pandoc#10362] [Q&A]
Hi, I am converting a LaTeX document to docx (a journal only accepts that, unfortunately). Pandoc 3.5 does a great job for the math, which I love.

I think (is that correct?) that the best tool to convert references is pandoc-crossref. I am calling it with `--filter pandoc-crossref -M autoEqnLabels=true`.

Unfortunately, equation numbers like `\ref{eq:something}` turn to hyperlinks `[eq:something]`. Is there a way to get numbers there? Ideally for figures, too.

If docx does not support this, odt is also an option, and so is rtf (as long as it can handle math).

**Comment by tarleb** [comment]
You could try with [querverweis](https://github.com/tarleb/querverweis) and possibly [rivnjannja](https://github.com/tarleb/rivnjannja). No promises, as they are work in progress, but I'd still be happy about bug reports if you do try those.

**Comment by iandol** [comment]
Hi @tarleb, on a quick look I found it hard to understand the exact scope of querverweis, the examples mostly show HTML input, but it is not limited to that?

**Reply by tarleb** [reply]
My personal use for querverweis is to support authors who write Markdown, but it should also work with other inputs. Rivnjannja is currently limited to LaTeX *output*, but I'd be open to extend its scope.

About the docs: they are a bit of an experiment, as they are generated from the [perevirky](https://github.com/tarleb/perevirky) tests that are used to validate the filter. Suggestions for improvements on that front are welcome as well.

**Reply by iandol** [reply]
Thanks for a very nice filter!!! I suppose I would have found it easier to grok if the first examples were markdown, I was also a bit flummoxed by the #id in the table (it makes sense for html input, not markdown till I kept going to see your span solution). Here is some off-the-cuff introduction, take it as a suggestion:

Querverweis is a cross-referencing Lua filter for Pandoc that adds numbered cross-references in the text using standard inline links to `#id` properties for figures and section titles. It supports injecting `#id`s in cases where Pandoc does not natively support it (tables and equations). For figures and tables it can optionally generate labels in the output captions for formats like HTML. 

Consider a simple figure with link (`labels` option is enabled, see [below for details](#labels-add-labels-to-caption)):

```markdown
---
querverweis:
  labels: true
---
![A beautiful sunset.](sunset.jpg){#sunset}

See figure [](#sunset) for a sunset.
```

…would generate this HTML output:

```html
<figure id="sunset">
<img src="sunset.jpg" alt="A beautiful sunset." />
<figcaption><span class="caption-label">Figure 1 </span>A beautiful sunset.</figcaption>
</figure>
<p>See figure <a href="#sunset">1</a> for a sunset.</p>
```

**Reply by tpapp** [reply]
@tarleb: if

> Rivnjannja is currently limited to LaTeX output

then I don't understand how it is useful for LaTeX -> docx conversion (see the OP). did I miss something?

**Reply by tarleb** [reply]
@iandol Thanks a lot, I used that almost verbatim.

@tpapp: I mentioned it because "I'd be open to extend its scope". So if it looks useful but doesn't do what you need, then please raise an issue.
